src/pages/Scan.vue

```diff
@@ -85,10 +85,13 @@ export default {
                         // Add error message for invalid and fail codes.
                         if (
                             response.data.invalid.length > 0
-                            || response.data.fail.length > 0
                         ) {
                             this.showFail();
-                            this.errorMessage = "Please enter a valid code.";
+                            this.errorMessage = "The voucher code you entered is not valid. Please try again.";
+                        } else if (
+                            response.data.fail.length > 0
+                        ) {
+                            this.errorMessage = "The voucher code has you entered has previously been submitted. Please try again.";
                         } else {
```

Comment:
Those are really the same error - and should use the same message, rather than be repeated in 2 places. Shouldn't the Scan page and the Tap page use the same submit methods anyway?